Small improvements

gprodanov opened the pull request:
A few small improvements:
- add a `.gitignore` file
- add a `package-lock.json` file
- fix typo in readme
- fix a few stylistic inconsistencies in `index.js`

Commit:
Add a .gitignore file

Commit:
Fix typo in README.md

Commit:
Add a package-lock.json file

Commit:
Minor refactoring of index.js

dokov commented on README.md:
```diff
@@ -18,7 +18,7 @@ Even if you have idling disabled, your Flex Service will sometimes be stopped an
 
 ## Implementation
-This Flex Service uses the kafkajs module to consume and produce Kafka messages. It subscribes to a topis and then sends messages to this same topic every 1 second. The service also intercepts shutdown signals and exits gracefully, waiting for all messages that are already being processed to finish processing. A 500-2000 ms delay is introduced on purpose in the processing of the messages so that  ...
+This Flex Service uses the kafkajs module to consume and produce Kafka messages. It subscribes to topics and then sends messages to this same topic every 1 second. The service also intercepts shutdown signals and exits gracefully, waiting for all messages that are already being processed to finish processing. A 500-2000 ms delay is introduced on purpose in the processing of the messages so that g ...
```
```suggestion
This Flex Service uses the kafkajs module to consume and produce Kafka messages. It subscribes to a topic and then sends messages to this same topic every 1 second. The service also intercepts shutdown signals and exits gracefully, waiting for all messages that are already being processed to finish processing. A 500-2000 ms delay is introduced on purpose in the processing of the messages so that graceful shutdown can be easily understood and tested.
```

Commit:
Update README.md

Co-authored-by: Lyubomir Dokov <lyubomir.dokov@telerik.com>

Commit:
Graceful shutdown improvements

Trigger the Flex SDK shutdown process after long-running task processing uninitialization. Let process exit naturally, instead of using process.exit(). Document the shutdown process.

Commit:
Slow down processing of messages

For better visibility, let the queue empty more slowly. Improve logging on unhandled errors.
